File: app/src/main/java/in/engineerakash/covid19india/util/ConnectivityReceiver.java
package in.engineerakash.covid19india.util;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class ConnectivityReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {

        if (Helper.isNetworkAvailable(context))
            intent.putExtra("connected", true);
        else
            intent.putExtra("connected", false);

        ObservableObject.getInstance().updateValue(intent);
    }
}

File: app/src/main/java/in/engineerakash/covid19india/util/ConnectivityReceiver.kt
package `in`.engineerakash.covid19india.util

import `in`.engineerakash.covid19india.util.Helper.isNetworkAvailable
import `in`.engineerakash.covid19india.util.ObservableObject.Companion.instance
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class ConnectivityReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (isNetworkAvailable(context)) intent.putExtra(
            "connected",
            true
        ) else intent.putExtra("connected", false)
        instance.updateValue(intent)
    }

}